src/psyclone/tests/domain/lfric/lfric_types_test.py

```diff
@@ -0,0 +1,398 @@
+    ("NumberOfQrPointsInXyDataType", "LfricIntegerScalarDataType"),
+    ("NumberOfQrPointsInZDataType", "LfricIntegerScalarDataType"),
+    ("NumberOfQrPointsInFacesDataType", "LfricIntegerScalarDataType"),
+    ("NumberOfQrPointsInEdgesDataType", "LfricIntegerScalarDataType")])
+def test_specific_scalar_types(data_type_name, generic_type_name):
+    '''Test the generated specific scalar datatypes are created correctly.
+
+    '''
+    lfric_types = LFRicTypes()
+    lfric_datatype = lfric_types(data_type_name)()
+    assert isinstance(lfric_datatype, lfric_types(generic_type_name))
+
+
+def test_specific_scalar_types_init_args():
+    '''Test the generated specific scalar data types are created correctly
+    and work with positional and keyword arguments.
+
+    '''
+    lfric_types = LFRicTypes()
+    type_class = lfric_types("NumberOfDofsDataSymbol")
+    positional_init = type_class("name", "fs")
+    assert positional_init.fs == "fs"
+    keyword_init = type_class("name", fs="fs")
+    assert keyword_init.fs == "fs"
+
+
+# Specific scalar symbols
+@pytest.mark.parametrize("symbol_name, generic_symbol_name, attribute_map", [
+    ("CellPositionDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("MeshHeightDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfCellsDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfDofsDataSymbol", "LfricIntegerScalarDataSymbol", {"fs": "w3"}),
+    ("NumberOfUniqueDofsDataSymbol", "LfricIntegerScalarDataSymbol",
+     {"fs": "w2"}),
+    ("NumberOfFacesDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfEdgesDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfQrPointsInXyDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfQrPointsInZDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfQrPointsInFacesDataSymbol", "LfricIntegerScalarDataSymbol", {}),
+    ("NumberOfQrPointsInEdgesDataSymbol", "LfricIntegerScalarDataSymbol", {})])
+def test_specific_scalar_symbols(symbol_name, generic_symbol_name,
+                                 attribute_map):
+    '''Test the generated specific scalar symbols are
+    created correctly.
+
+    '''
+    lfric_types = LFRicTypes()
+    symbol = lfric_types(symbol_name)
+    generic_symbol = lfric_types(generic_symbol_name)
+    args = ["symbol"] + list(attribute_map.values())
+    lfric_symbol = symbol(*args)
+    assert isinstance(lfric_symbol, generic_symbol)
+    assert lfric_symbol.name == "symbol"
+    assert isinstance(lfric_symbol.interface, LocalInterface)
+    for attribute in attribute_map:
+        assert getattr(lfric_symbol, attribute) == attribute_map[attribute]
+    lfric_symbol = symbol(
+        *args, interface=ArgumentInterface(ArgumentInterface.Access.READ))
+    assert isinstance(lfric_symbol.interface, ArgumentInterface)
+    assert lfric_symbol.interface.access == ArgumentInterface.Access.READ
+
+
+# Specific scalar datatypes
+@pytest.mark.parametrize(
```

Comment:
I'm assuming this is all just copied from `lfric_psyir_test.py`?

Reply:
Yes.